x/uibc/msg.go

```diff
@@ -26,21 +25,24 @@ func (msg *MsgGovUpdateQuota) String() string {
 
 // ValidateBasic implements Msg
 func (msg *MsgGovUpdateQuota) ValidateBasic() error {
+	var errs []error
 	if err := checkers.Proposal(msg.Authority, msg.Description); err != nil {
-		return err
+		errs = append(errs, err)
 	}
 
-	if msg.Total.IsNil() || !msg.Total.IsPositive() {
-		return sdkerrors.ErrInvalidRequest.Wrap("total quota must be positive")
-	}
-	if msg.PerDenom.IsNil() || !msg.PerDenom.IsPositive() {
-		return sdkerrors.ErrInvalidRequest.Wrap("quota per denom must be positive")
-	}
+	errs = checkers.DecPositive(msg.Total, "total quota", errs)
```

Comment:
To keep this "one line per check" feature while implementing the comment above, you could make

```
errs = checkers.AppendErr(errs, checkers.DecPositive(msg.Total, "total quota")
```

where `AppendErr` appends an error if non-nil

Reply:
I think we should use more often the aggregated errors which came to `go v1.20`. So adding `checkers.AppendErr` is additional code we don't need.

Reply:
`errors.Join` works too

Reply:
yes, that's why we can have one list of errors, and then do `errors.Join`. It reduces the code, without adding additional layers.

Reply:
so alternative would be, instead of
```errs = checkers.XXX(...)
errs = checkers.XXX(...)
errs = checkers.XXX(...)
errs = checkers.XXX(...)
errs = checkers.XXX(...)
return errors.Join(errs...)
```
we would do
```err1 = checkers.XXX(...)
err2 = checkers.XXX(...)
err3 = checkers.XXX(...)
err4 = checkers.XXX(...)
err5 = checkers.XXX(...)
return errors.Join(err1, err2, err3, err4, err5)
```

The problem here is that we can  forget one of the err1, err2, ....

Reply:
However if we would do
```
err = errors.Join(err, checkers.XXX(...))
err = errors.Join(err, checkers.XXX(...))
err = errors.Join(err, checkers.XXX(...))
err = errors.Join(err, checkers.XXX(...))
err = errors.Join(err, checkers.XXX(...))
return err
```
The code is more bloated, and creates lof of wrapping, so we have many nested errors.

Reply:
In my opinion this is fine because we don’t know how many errors we will get 
```go
errs = checkers.XXX(...) -> append errors
errs = checkers.XXX(...)
errs = checkers.XXX(...)
errs = checkers.XXX(...)
errs = checkers.XXX(...) -> append errors
return  errs 
```

Reply:
> However if we would do
> 
> ```
> err = errors.Join(err, checkers.XXX(...))
> err = errors.Join(err, checkers.XXX(...))
> err = errors.Join(err, checkers.XXX(...))
> err = errors.Join(err, checkers.XXX(...))
> err = errors.Join(err, checkers.XXX(...))
> return err
> ```
> 
> The code is more bloated, and creates lof of wrapping, so we have many nested errors.

This looks better to me

Reply:
But here if `checkers.XXX(...)` is nil , it will append `nil` to errors @kosegor

Reply:
Join actually ignores nil (no-op), so it's fine

Reply:
but `errors.Join` will make unnecessary wrappers, so under the hood we will have `wrapper{ wrapper { wrapper {...} } }`